packages/rum/src/domain/record/elementsScrollPositions.ts

```diff
@@ -1,27 +1,19 @@
-import { addTelemetryDebug } from '@datadog/browser-core'
-
 export type ElementsScrollPositions = ReturnType<typeof createElementsScrollPositions>
 export type ScrollPositions = { scrollLeft: number; scrollTop: number }
 
 export function createElementsScrollPositions() {
   const scrollPositionsByElement = new WeakMap<Element, ScrollPositions>()
-  let documentScrollingElement: Element | null
   return {
     set(element: Element | Document, scrollPositions: ScrollPositions) {
-      if (element === document && !documentScrollingElement) {
-        documentScrollingElement = tryToFindScrollingElement(scrollPositions)
-        if (!documentScrollingElement) {
-          return
-        }
-      }
-      try {
-        scrollPositionsByElement.set(
-          element === document ? documentScrollingElement! : (element as Element),
-          scrollPositions
-        )
-      } catch (e) {
-        addTelemetryDebug(`invalid element: ${String(element)}`)
+      if (element === document && !document.scrollingElement) {
+        // cf https://drafts.csswg.org/cssom-view/#dom-document-scrollingelement,
+        // in some cases scrolling elements can not be defined, we don't support those for now
+        return
       }
+      scrollPositionsByElement.set(
+        element === document ? document.scrollingElement! : (element as Element),
+        scrollPositions
+      )
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏I feel writing it like that would be a tiny bit clearer, but either is fine

```suggestion
      if (element === document) {
        if (!document.scrollingElement) {
          // cf https://drafts.csswg.org/cssom-view/#dom-document-scrollingelement,
          // in some cases scrolling elements can not be defined, we don't support those for now
          return
        }
        element = document.scrollingElement
      }
      scrollPositionsByElement.set((element as Element), scrollPositions)
```